lib/oneview-sdk/rest.rb

```diff
@@ -121,6 +123,75 @@ def rest_delete(path, options = {}, api_ver = @api_version)
       rest_api(:delete, path, options, api_ver)
     end
 
+    # Uploads a file to a specific uri
+    # @param [String] file_path
+    # @param [String] uri The uri starting with "/"
+    # @param [Hash] body_params The params to append to body of http request. Default is {}.
+    # @option body_params [String] 'name' The name to show (when resource accepts a name)
+    # @param [Integer] timeout The number of seconds to wait for completing the request. Default is 300.
+    # @return [OneviewSDK::Resource] if the upload was sucessful, return a Resource object
+    def upload_file(file_path, uri, body_params = {}, timeout = READ_TIMEOUT)
+      raise NotFound, "ERROR: File '#{file_path}' not found!" unless File.file?(file_path)
+      options = {
+        'Content-Type' => 'multipart/form-data',
+        'X-Api-Version' => @api_version.to_s,
+        'auth' => @token
+      }
+      url = URI.parse(URI.escape("#{@url}#{uri}"))
+
+      File.open(file_path) do |file|
+        name_to_show = body_params.delete('name') || body_params.delete(:name)
+        name_to_show = name_to_show ? name_to_show + File.extname('/tmp/artifact_bundle.zip') : File.basename(file_path)
```

Comment:
Sorry, my intention is use `File.extname(file_path)` because I want just change the name of file (using the same extension) if new name is present.
But I will validate if new name has already the extension too. =)
I shouldn't use '.zip' directly because the upload_file method is generic, and user can be do upload of a file with another extension.